File: Muc_Do_3_Kha_Kho/Bai15_TongCacTong/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI15.INP', 'r')
        sys.stdout = open('BAI15.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    arr = [int(x) for x in data[1:1+n]]
    arr.sort()
    ans = 0
    for i in range(n):
        ans += arr[i] * (2 * i - n + 1)
    print(ans)
solve()

File: Muc_Do_3_Kha_Kho/Bai15_TongCacTong/Solution.cpp
#include <bits/stdc++.h>
using namespace std;
// Bai 15: Tong khoang cach (Manhattan 1D)
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI15.INP", "r")) {
        freopen("BAI15.INP", "r", stdin);
        freopen("BAI15.OUT", "w", stdout);
    }
    int n;
    if (!(cin >> n)) return 0;
    vector<long long> a(n);
    for (int i = 0; i < n; ++i) cin >> a[i];
    sort(a.begin(), a.end());
    
    long long ans = 0;
    for (int i = 0; i < n; ++i) {
        ans += a[i] * (2LL * i - n + 1);
    }
    cout << ans << "\n";
    return 0;
}
